numpy/core/numerictypes.py

```diff
@@ -181,8 +181,7 @@ def maximum_sctype(t):
     if g is None:
         return t
     t = g
-    bits = _bits_of(t)
-    base = _kind_to_stem[dtype(t).kind]
+    base = _kind_name(dtype(t))
```

Comment:
Same remark about this now raising `RuntimeError` instead of `KeyError` for weird dtypes.